Source/ARTDevicePushDetails.h

```diff
@@ -1,14 +1,14 @@
-#import <Foundation/Foundation.h>
+#import <Ably/ARTTypes.h>
 
 @class ARTErrorInfo;
 
 NS_ASSUME_NONNULL_BEGIN
 
 @interface ARTDevicePushDetails : NSObject
 
-@property (nonatomic) NSMutableDictionary<NSString *, NSString *> *recipient;
-@property (nullable, nonatomic) NSString *state;
+@property (nonatomic) ARTPushState state;
```

Comment:
I think this change (string to enum) can be a separate thing, which we should do regardless of how we solve the `LocalDevice` issue. I've created an issue for it: https://github.com/ably/ably-cocoa/issues/1264.